simbad/util/tests/test_pdb_util.py

```diff
@@ -78,6 +68,113 @@ def test_molecular_weight_3(self):
+
+    def test_nchains_2(self):
+        """Test case for PdbStructure.nchains"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        data = pdb_struct.nchains
+        reference_data = 2
+
+        self.assertEqual(data, reference_data)
+
+    def test_nres_1(self):
+        """Test case for PdbStructure.nres"""
+        input_model = os.path.join(EXAMPLE_DIR, "toxd", "toxd.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        data = pdb_struct.nres
+        reference_data = 59
+
+        self.assertEqual(data, reference_data)
+
+    def test_nres_2(self):
+        """Test case for PdbStructure.nres"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        data = pdb_struct.nres
+        reference_data = 192
+
+        self.assertEqual(data, reference_data)
+
+    def test_keep_first_chain(self):
+        """Test case for PdbStructure.keep_first_chain_only"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.keep_first_chain_only()
+        data = pdb_struct.nchains
+        reference_data = 1
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_chain_by_idx_1(self):
+        """Test case for PdbStructure.select_chain_by_idx"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.select_chain_by_idx(0)
+        data = pdb_struct.structure[0][0].name
+        reference_data = 'A'
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_chain_by_idx_2(self):
+        """Test case for PdbStructure.select_chain_by_idx"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.select_chain_by_idx(1)
+        data = pdb_struct.structure[0][0].name
+        reference_data = 'B'
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_chain_by_id_1(self):
+        """Test case for PdbStructure.select_chain_by_id"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.select_chain_by_id('A')
+        data = pdb_struct.structure[0][0].name
+        reference_data = 'A'
+
+        self.assertEqual(data, reference_data)
+
+    def test_select_chain_by_id_2(self):
+        """Test case for PdbStructure.select_chain_by_id"""
+        input_model = os.path.join(EXAMPLE_DIR, "rnase", "rnase.pdb")
+        pdb_struct = PdbStructure.from_file(input_model)
+        pdb_struct.select_chain_by_id('B')
+        data = pdb_struct.structure[0][0].name
+        reference_data = 'B'
+
+        self.assertEqual(data, reference_data)
+
```

Comment:
In all of these tests you should also check that the `pdb_struct.nchains` equals 1, otherwise you don't know if only the desired chain was selected.